src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,10 +33,24 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
+
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        processAddress(trimmedAddress);
+    }
+
+    /**
+     * processes different components of address(block,street,unit,postal code)
```

Comment:
The start of the header comment does not follow the convention (it should start with a capital letter).